app/scripts/controller-init/messengers/accounts/account-tree-controller-messenger.ts

```diff
@@ -0,0 +1,60 @@
+import { Messenger } from '@metamask/base-controller';
+import {
+  AccountsControllerAccountAddedEvent,
+  AccountsControllerAccountRemovedEvent,
+  AccountsControllerListMultichainAccountsAction,
+} from '@metamask/accounts-controller';
+import { GetSnap as SnapControllerGet } from '@metamask/snaps-controllers';
+import { KeyringControllerGetStateAction } from '@metamask/keyring-controller';
+
+type Actions =
+  | AccountsControllerListMultichainAccountsAction
+  | SnapControllerGet
+  | KeyringControllerGetStateAction;
+
+type Events =
+  | AccountsControllerAccountAddedEvent
+  | AccountsControllerAccountRemovedEvent;
+
+export type AccountTreeControllerMessenger = ReturnType<
+  typeof getAccountTreeControllerMessenger
+>;
+
+/**
+ * Get a restricted messenger for the account wallet controller. This is scoped to the
+ * actions and events that this controller is allowed to handle.
+ *
+ * @param messenger - The controller messenger to restrict.
+ * @returns The restricted controller messenger.
+ */
+export function getAccountTreeControllerMessenger(
+  messenger: Messenger<Actions, Events>,
+) {
+  return messenger.getRestricted({
+    name: 'AccountTreeController',
+    allowedEvents: [
+      'AccountsController:accountAdded',
+      'AccountsController:accountRemoved',
+    ],
+    allowedActions: [
+      'AccountsController:listMultichainAccounts',
+      'SnapController:get',
+      'KeyringController:getState',
+    ],
```

Comment:
Why do we use the raw string instead of importing them from the specific packages?

Reply:
Good question, I blindly followed the pattern from other controllers, but TBH we are mixing this in other areas too. (would be great to always follow the same pattern at some point 😅).

That being said, those are validated statically, so we cannot reference an invalid actions/events here.

I'll keep it as is for now.